src/codeflare_sdk/cluster/config.py

```diff
@@ -63,21 +70,54 @@ def __post_init__(self):
             )
         self._memory_to_string()
         self._str_mem_no_unit_add_GB()
+        self._gpu_to_resource()
+        self._cpu_to_resource()
+        self._memory_to_resource()
 
     def _str_mem_no_unit_add_GB(self):
         if isinstance(self.head_memory, str) and self.head_memory.isdecimal():
             self.head_memory = f"{self.head_memory}G"
-        if isinstance(self.min_memory, str) and self.min_memory.isdecimal():
-            self.min_memory = f"{self.min_memory}G"
-        if isinstance(self.max_memory, str) and self.max_memory.isdecimal():
-            self.max_memory = f"{self.max_memory}G"
+        if (
+            isinstance(self.worker_memory_requests, str)
+            and self.worker_memory_requests.isdecimal()
+        ):
+            self.worker_memory_requests = f"{self.worker_memory_requests}G"
+        if (
+            isinstance(self.worker_memory_limits, str)
+            and self.worker_memory_limits.isdecimal()
+        ):
+            self.worker_memory_limits = f"{self.worker_memory_limits}G"
 
     def _memory_to_string(self):
         if isinstance(self.head_memory, int):
             self.head_memory = f"{self.head_memory}G"
-        if isinstance(self.min_memory, int):
-            self.min_memory = f"{self.min_memory}G"
-        if isinstance(self.max_memory, int):
-            self.max_memory = f"{self.max_memory}G"
+        if isinstance(self.worker_memory_requests, int):
+            self.worker_memory_requests = f"{self.worker_memory_requests}G"
+        if isinstance(self.worker_memory_limits, int):
+            self.worker_memory_limits = f"{self.worker_memory_limits}G"
+
+    def _gpu_to_resource(self):
+        if self.head_gpus:
+            warnings.warn("head_gpus is being deprecated, use num_head_gpus")
+            self.num_head_gpus = self.head_gpus
+        if self.num_gpus:
+            warnings.warn("num_gpus is being deprecated, use num_worker_gpus")
+            self.num_worker_gpus = self.num_gpus
+
+    def _cpu_to_resource(self):
+        if self.min_cpus:
+            warnings.warn("min_cpus is being deprecated, use worker_cpu_requests")
+            self.worker_cpu_requests = self.min_cpus
+        if self.max_cpus:
+            warnings.warn("max_cpus is being deprecated, use worker_cpu_limits")
+            self.worker_cpu_limits = self.max_cpus
+
+    def _memory_to_resource(self):
+        if self.min_memory:
+            warnings.warn("min_memory is being deprecated, use worker_memory_requests")
+            self.worker_memory_requests = self.min_memory
+        if self.max_memory:
+            warnings.warn("max_memory is being deprecated, use worker_memory_limits")
+            self.worker_memory_limits = self.max_memory
```

Comment:
The `G` unit is not added when using the deprecating parameters for the memory:
 ```
         resources:
            limits:
              cpu: 1
              memory: 2
              nvidia.com/gpu: 1
            requests:
              cpu: 1
              memory: 2
              nvidia.com/gpu: 1
```
You could reformat as such i.e.:
 `self.worker_memory_limits = f"{self.max_memory}G"`

Reply:
added this and yes it fixed it in my latest commit, thanks Christian